[60667] Automatic scheduling mode: make it possible to enter Finish date by direct input or by clicking on the mini calendar

HDinger opened the pull request:
# Ticket
https://community.openproject.org/projects/openproject/work_packages/60667/activity

# What are you trying to accomplish?
* Allow editing the finish date in automatic scheduling mode

## Screenshots
<img width="631" alt="Bildschirmfoto 2025-03-24 um 14 48 14" src="https://github.com/user-attachments/assets/50b8ac87-5158-4334-8464-bcbcb3ea9966" />

# Merge checklist

- [ ] Added/updated tests

HDinger committed:
Allow editing the finish date in automatic scheduling mode

cbliard reviewed (changes requested):
It works better for the case pinpointed before, but now it's impossible to select the finish date from the calendar.

[setting end date not works.webm](https://github.com/user-attachments/assets/e7767bf3-0433-4ccb-a488-02fc24f17903)

.

Changing the finish date through the input fields works.

[cannot set due date by clicking calendar.webm](https://github.com/user-attachments/assets/d30266d0-6267-4a47-b520-b703d2056518)

cbliard reviewed (changes requested):
That works well!
Especially the way it handles duration or finished date having been entered last and being kept when checking/unchecking "Working days only". Really shiny 🥇 

I have only found one minor issue: in a milestone work package type in automatic scheduling, the calendar is enabled and it's possible to change the date. It should be disabled as there is only one date to be picked and it is automatically set.

HDinger commented:
> I have only found one minor issue: in a milestone work package type in automatic scheduling, the calendar is enabled and it's possible to change the date. It should be disabled as there is only one date to be picked and it is automatically set.

Good catch @cbliard , I fixed the issue you mentioned 👍

cbliard reviewed (approved):
Nice! I could not find any issues this time.
Good job.

psatyal commented:
Looks brilliant! Most of it is now all good.

Just found a little bug: 

**Start:** Auto scheduled. Has start and finish date.
**Action**: Open date picker by clicking on duration. Then click on a date in the future.
**Currenty result**: The dates/range are lost.
**Expected behaviour**: The click changes the finish date (and duration is updated).

Every else works excellently.

Cheers
Parimal

cbliard commented:
Nice catch @psatyal

HDinger committed:
When duration is changed in automatic mode, the finish date has to be adjusted

HDinger committed:
Add test cases for making the due date editable in automatic mode

cbliard committed:
Fir wrong test conditions

The setup was incorrect: the successor's dates were identical to the predecessor's dates which is not possible, and the milestone had start date different from due date which is not possible either.

Finally the second test was wrong too because when a successor milestone is
automatically scheduled, the date can't be changed and the date picker
is fully disabled.

cbliard committed:
Fix spec
